src/robot.coffee

```diff
@@ -247,6 +264,28 @@ class Robot
   # Returns nothing.
   # Returns before executing callback
   receive: (message, cb) ->
+    # When everything is finished (down the middleware stack and back up),
+    # pass control back to the robot
+    receiveMiddlewareDone = () ->
+      if cb?
+        process.nextTick -> cb true
+
+    @middleware.receive.execute(
+      {response: new Response(this, message)}
```

Comment:
> This means, for example, that any attempts to inject custom response interception (response middleware) via receive middleware are going to fail.

And I just realized this statement is completely stupid. A throwaway Response object is no different from not having a Response object with respect to response middleware. Nevermind...